packages/cozy-harvest-lib/src/connections/files.js

```diff
@@ -0,0 +1,39 @@
+const FILES_DOCTYPE = 'io.cozy.files'
+
+/**
+ * Creates a directory from a given path
+ * @param  {Object}  client CozyClient
+ * @param  {string}  path   Directory path
+ * @return {Object}         Directory attributes
+ */
+export const createDirectoryByPath = async (client, path) => {
+  const { data } = await client
+    .collection(FILES_DOCTYPE)
+    .createDirectoryByPath(path)
+  return data
+}
+
+/**
+ * Retrieves a directory from its path
+ * @param  {Object}  client CozyClient
+ * @param  {string}  path   Directory path
+ * @return {Object}        Created io.cozy.files document
+ */
+export const fetchDirectoryByPath = async (client, path) => {
+  let response
+  try {
+    response = await client.collection(FILES_DOCTYPE).statByPath(path)
+  } catch (error) {
+    if (error && error.status === 404) return null
+    throw new Error(error.message)
+  }
+
+  return response.data
+}
```

Comment:
It seems that the two above methods have the same skeleton. I wonder if something like below could serve ?

```js
/* Decorates a function so that the first argument (client) becomes the collection */
const collectionMethod = doctype => collectionFn => (client, ...restArgs) => {
  const col = client.collection(doctype)
  return collectionFn(client, ...restArgs)
}

const fileMethod = collectionMethod(FILES_DOCTYPE)

export const createDirectoryByPath = fileMethod(async (files, path) => {
 const { data } = await files.createDirectoryByPath(path)
 return data
})

 export const softStatByPath = fileMethod(async (files, path) => {
  try {
    const { data } = files.statByPath(path)
    return data
  } catch (error) {
    if (error && error.status === 404) return null
    throw new Error(error.message)
  }
}
```

Reply:
I think it's worth refactoring those things since otherwise we are basically rewriting stackClient.

Reply:
The pattern is not repeated anymore, but it's a nice helper to keep for Cozy-Client and future Models I think. 👍